cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -692,6 +694,54 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 				ImageName: args.Targets[0].ImageName,
 			}))
 
+			osbuildJobResult.Success = true
+			osbuildJobResult.UploadStatus = "success"
+		case *target.KojiTargetOptions:
+			kojiServerURL, err := url.Parse(options.Server)
+			if err != nil {
+				osbuildJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorInvalidTargetConfig, fmt.Sprintf("failed to parse Koji server URL: %v", err))
+				return nil
+			}
+
+			creds, exists := impl.KojiServers[kojiServerURL.Hostname()]
+			if !exists {
+				osbuildJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorInvalidTargetConfig, fmt.Sprintf("Koji server has not been configured: %s", kojiServerURL.Hostname()))
+				return nil
+			}
+
+			kojiTransport := koji.CreateKojiTransport(impl.KojiRelaxTimeoutFactor)
+
+			kojiAPI, err := koji.NewFromGSSAPI(options.Server, &creds, kojiTransport)
+			if err != nil {
+				osbuildJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorInvalidTargetConfig, fmt.Sprintf("failed to authenticate with Koji server %q: %v", kojiServerURL.Hostname(), err))
+				return nil
+			}
+			logWithId.Infof("[Koji] 🔑 Authenticated with %q", kojiServerURL.Hostname())
+			defer func() {
+				err := kojiAPI.Logout()
+				if err != nil {
+					logWithId.Warnf("[Koji] logout failed: %v", err)
+				}
+			}()
+			file, err := os.Open(path.Join(outputDirectory, exportPath, args.ImageName))
+			if err != nil {
+				osbuildJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorKojiBuild, fmt.Sprintf("failed to open the image for reading: %v", err))
+				return nil
+			}
+			defer file.Close()
+
+			logWithId.Info("[Koji] ⬆ Uploading the image")
+			imageHash, imageSize, err := kojiAPI.Upload(file, options.UploadDirectory, options.Filename)
+			if err != nil {
+				osbuildJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorKojiBuild, err.Error())
```

Comment:
I agree. I didn't change it, because I didn't want to unexpectedly skew any statistics that we may be tracking for Koji builds, since this is what `osbuild-koji` is using
https://github.com/osbuild/osbuild-composer/blob/9bff4a4f0fc41224361fd8fbb687be47a0dc8be2/cmd/osbuild-worker/jobimpl-osbuild-koji.go#L165

But if it won't cause any issues, lets change it.